CRM/Utils/Token.php

```diff
@@ -1700,6 +1718,93 @@ public static function replaceMultipleContributionTokens($separator, $str, &$con
   }
 
   /**
+   * Get replacement strings for any event tokens
+   * @param string $token
+   * @param array $event an api result array for a single event
+   * @param bool $escapeSmarty
+   * @return string token replacement
+   */
+  public static function getEventTokenReplacement($token, $event, $escapeSmarty = FALSE) {
+    $entity = 'event';
+    self::_buildEventTokens();
+
+    $params = array('entity_id' => $event['id'], 'entity_table' => 'civicrm_event');
+
+    switch ($token) {
+      case 'balance':
+        $info = CRM_Contribute_BAO_Contribution::getPaymentInfo($params['entity_id'], 'event');
+        $balancePay = CRM_Utils_Array::value('balance', $info);
+        $balancePay = CRM_Utils_Money::format($balancePay);
+        $value = $balancePay;
+        break;
+
+      case 'title':
+        $value = $event['event_title'];
+        break;
+
+      case 'end_date':
+      case 'start_date':
+        $value = CRM_Utils_Date::customFormat($event[$token], "%d/%m/%Y");
+        break;
+
+      case 'type':
+        $params = array(
+          'option_group_id' => 14,
+          'value' => $event['event_type_id'],
+          'return' => 'label',
+        );
+        $value = civicrm_api3('OptionValue', 'getvalue', $params);
+        break;
+
+      case 'location':
+        $location = CRM_Core_BAO_Location::getValues($params, TRUE);
+        foreach ($location['address'] as $address) {
+          $value = $address['display_text'];
+        }
+        break;
+
+      case 'info_url':
+        $value = CIVICRM_UF_BASEURL . 'civicrm/event/info?reset=1&id=' . $event['id'];
+        break;
+
+      case 'registration_url':
+        $value = CIVICRM_UF_BASEURL . 'civicrm/event/register?reset=1&id=' . $event['id'];
+        break;
+
+      case 'fee_amount':
```

Comment:
 Does 'fee_amount' means event fee amount or total amount paid by the participant. If ’fee_amount' means event fee amount then it does not display correctly. 
For ex. Find Participant -> Create PDF from any sample "Rain-forest Cup Youth Soccer Tournament"-> Display Contribution Amount: $ 1.00